frame/contracts/src/exec.rs

```diff
@@ -369,16 +391,16 @@ where
 		gas_meter: &mut GasMeter<T>,
 		code_hash: &CodeHash<T>,
 		input_data: Vec<u8>,
-	) -> Result<(T::AccountId, ExecReturnValue), DispatchError> {
+	) -> Result<(T::AccountId, ExecReturnValue), ExecError> {
 		if self.depth == self.config.max_depth as usize {
-			Err("reached maximum depth, cannot instantiate")?
+			Err(Error::<T>::MaxCallDepthReached)?
 		}
 
 		if gas_meter
 			.charge(self.config, ExecFeeToken::Instantiate)
 			.is_out_of_gas()
 		{
-			Err("not enough gas to pay base instantiate fee")?
+			Err(Error::<T>::OutOfGas)?
```

Comment:
That's unfortunate that we lose some information here and above. In the string based error messages we differentiate between call and instantiate.

Reply:
Yes. But I prefer to have at least some kind of information in the UI and block explorers when this common error occurs. Of course we could split that into multiple errors but I think that will get out of hand rather quickly. We need to come up with a better solution for actual debugging by developers anyways. For the user it is not really interesting where it ran out of gas. 

That said, gas charged from exec.rs will go away soon anyways. All `ext_*` functions will be benchmarked for their worst case and include the costs for things that happen in exec.